src/solvers/floatbv/float_utils.cpp

```diff
@@ -136,6 +136,32 @@ bvt float_utilst::to_integer(
   return result;
 }
 
+bvt float_utilst::round_to_integral(const bvt &src)
+{
+  PRECONDITION(src.size() == spec.width());
+
+  // Zero? NaN? Infinity?
+  auto unpacked = unpack(src);
+  auto is_special = prop.lor({unpacked.zero, unpacked.NaN, unpacked.infinity});
+
+  // add 2^f, where f is the number of fraction bits,
+  // by adding f to the exponent
+  auto magic_number = ieee_floatt{
+    spec, ieee_floatt::rounding_modet::ROUND_TO_ZERO, power(2, spec.f)};
+
+  auto magic_number_bv = build_constant(magic_number);
+  magic_number_bv.back() = src.back(); // copy sign bit
+
+  auto tmp1 = add_sub(src, magic_number_bv, false);
+
+  auto tmp2 = add_sub(tmp1, magic_number_bv, true);
```

Comment:
If `ieee_floatt::round_to_integral` is to be trusted, shouldn't the sign bit be used instead of hard-coding `false` and `true`?

Reply:
Note line 153 -- this variant tweaks the sign bit of the magic constant instead.